lib/controller.js

```diff
@@ -641,7 +642,114 @@ controller.tesselFirmwareVerion = function(opts) {
         });
     });
 };
+/*
+Usage:
+controller.menu.create(tessels)
+        .then(function(tessel) {
+          return resolve(tessel);            
+        });
+*/
+controller.menu = {
+  create: function(tessels, opts, title) {
+    return new Promise(function(resolve) {
+      var rtm = new Menu({
+        //  width: process.stdout.columns - 4,
+        width: 55,
+        x: 1,
+        y: 2,
+        bg: 'red'
+      });
 
+      // create menu entries
+
+      controller.menu.make(tessels, rtm, title)
+        .then(controller.menu.show(rtm))
+        .then(function(index) {
+          resolve(tessels[index]);
+        });
+    });
+  },
+  make: function(tessels, rtm, title) {
+    return new Promise(function(resolve, reject) {
+      rtm.reset();
+      if (!title) {
+        title = 'MENU';
+      }
+      rtm.once('select', function(label, index) {
+        process.stdin.setRawMode(false);
+        process.stdin.end();
+        // Identify the Exit command by first letter (all other entries start with numbers/index)
+        if (label[0] !== 'E') {
+          resolve(index);
+        } else {
+          // going to clear screen and calling exit to resolve promise
+          controller.menu.clear()
+            .then(process.exit());
+        }
+      });
+      rtm._draw();
+      rtm.write(title + '\n');
+      rtm.write('                                                \n');
+      async.forEachOf(tessels, function addItem(tessel, index, done) {
+          if (tessel.lanConnection) {
+
+            if (!tessel.lanConnection.authorized) {
+              rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
+            } else {
+              rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
+            }
+
+          } else if (tessel.usbConnection) {
+            if (!tessel.usbConnection.authorized) {
+              rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
+            } else {
+              rtm.add(index + ') ' + tessel.name + ': [USB] (authorized) \n');
+            }
+          } else {
+            reject('Type of Tessel unkonwn: ', tessel);
+          }
+          done();
+        },
+        function closed(err) {
+          if (err) {
+            logs.warn(err);
```

Comment:
Just pass the error string into `reject` and it will be WARNed automatically by `tessel-2.js`.